### PR TITLE
Consensus: Turn CBlockIndex::GetMedianTimePast into independent function

### PR DESCRIPTION
**Pull request by jtimon**
ContextualCheckBlockHeader() [which is needed to verify a block header] depends on CBlockIndex::GetMedianTimePast but the goal is for verify block header is to be independent from chain.o and CBlockIndex C++ class [which cannot be exposed in the consensus C API].
This is also contained in #5995.

### CODE REVIEW

**Comment by jtimon** [comment]
Needed rebase.

**Comment by jtimon** [comment]
Needed rebase.

**Comment by sipa** [comment]
utACK

**Comment by sipa** [comment]
Needs rebase.

**Comment by jtimon** [comment]
@sipa I know, but I'm waiting for #6051 to be merged since it will require another rebase after that.

**Comment by jtimon** [comment]
Rebased

File: src/consensus/blockverify.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "consensus/consensus.h"
+
+#include "chain.h"
+
+#include <algorithm>
```
Comment (petertodd) [inline src/consensus/blockverify.cpp]: Nit: tailing whitespace

**Comment by petertodd** [comment]
Modulo nit, (somewhat tested) ACK.

**Comment by jtimon** [comment]
@petertodd 's nit fixed.

**Comment by btcdrak** [comment]
utACK

**Comment by jtimon** [comment]
Closing for now.

**Comment by jtimon** [comment]
Rebased and reopened.

**Comment by jtimon** [comment]
Rebased on top of #6591. Still had merging conflicts so rebased on top of master after that.

**Comment by dcousens** [comment]
utACK for c85ded0 after nit is addressed

**Comment by jtimon** [comment]
Closing in favor of #6672 .

**Pull request closed without merging**